### PR TITLE
Issue #8657: Modification of javadoc example markers to support metadata generation - regexp

### PR DESCRIPTION
**Pull request by gaurabdg**
#8657

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java [hunk 1]
```diff
@@ -122,6 +122,9 @@
  * </li>
  * </ul>
  * <p>
+ * To configure the check:
+ * </p>
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java:126]: this one a bit more complicated ...

please add `To configure the default check` and add default example as we did gsoc-first-issue.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java [hunk 2]
```diff
@@ -100,7 +100,9 @@
  * Type is {@code java.lang.String[]}.
  * Default value is {@code ""}.</li>
  * </ul>
- *
+ * <p>
+ * To configure the check:
+ * </p>
  * <p>
  * To configure the check to report file names that contain a space:
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java:107]: `To configure the default check that report ...`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java [hunk 3]
```diff
@@ -68,6 +68,9 @@
  * </li>
  * </ul>
  * <p>
+ * To configure the check:
+ * </p>
+ * <p>
  *   To configure the check with default values:
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java:74]: `To configure the default check:`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java (hunk 1 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java:126]: There are various example configs already present in the lines below. This line is just used for signifying the start of examples section.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java (hunk 2 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java:107]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java (hunk 3 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java:74]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java:126]: ok, let it be, we will address it later

**Pull request merged**